app/Jobs/ProcessMediaWikiJobsJob.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace App\Jobs;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Log;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class ProcessMediaWikiJobsJob implements ShouldQueue, ShouldBeUnique
+{
+    use InteractsWithQueue, Queueable;
+
+    private string $wikiDomain;
+
+    public function __construct ( string $wikiDomain )
+    {
+        $this->wikiDomain = $wikiDomain;
+    }
+
+    public function uniqueId(): string
+    {
+        return $this->wikiDomain;
+    }
+
+    public function handle (Client $kubernetesClient): void
+    {
+        $mwPod = $kubernetesClient->pods()->setFieldSelector([
+            'status.phase' => 'Running'
+        ])->setLabelSelector([
+            'app.kubernetes.io/name' => 'mediawiki',
+            'app.kubernetes.io/component' => 'app-backend'
+        ])->first();
+
+        if (!$mwPod) {
+            $this->fail(
+                new \RuntimeException(
+                    'Unable to find a running MediaWiki pod in the cluster, '.
+                    'cannot continue.'
+                )
+            );
+            return;
+        }
+
+        $mwPod = $mwPod->toArray();
+
+        $jobSpec = new KubernetesJob([
+            'metadata' => [
+                'generateName' => 'run-all-mw-jobs-'
+            ],
+            'spec' => [
+                'ttlSecondsAfterFinished' => 60 * 60,
```

Comment:
Is it a sane idea to keep job objects around for an hour only?

Reply:
I think it would be fine